agent/api/container.go

```diff
@@ -235,6 +237,34 @@ func (c *Container) GetKnownExitCode() *int {
 	return c.knownExitCode
 }
 
+// SetRegistryAuthCredentials will set the credentials for pulling image from ecr
+func (c *Container) SetRegistryAuthCredentials(credential credentials.IAMRoleCredentials) error {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	if c.RegistryAuthentication == nil || c.RegistryAuthentication.ECRAuthData == nil {
+		return fmt.Errorf("set ecr pull credentials failed, no registry auth data for container")
```

Comment:
I don't think we need the `"set ecr pull credentials failed"` phrase in the error message. Can you replace that with something like `"ecr pull credentials: ..."` here and below?